pkgs/build-support/cc-wrapper/setup-hook.sh

```diff
@@ -69,10 +69,12 @@ ccWrapper_addCVars () {
     getHostRoleEnvHook
 
     if [ -d "$1/include" ]; then
+        (! echo $NIX_CFLAGS_COMPILE | grep $1 > /dev/null) &&
```

Comment:
Does this implementation change the search order or not? I guess the question is if (previously) the duplicates were found first, or the originals.

Reply:
The scenario I'm thinking of is two packages, named a and b, providing the same named header file, and then they are added as build inputs in the following order: [ a b a ]. We should ensure this PR doesn't change the search order.

Reply:
For stability, sure, but I think relying (knowingly or unknowingly) on the search order is a bit hacky, and should be fixed in the packages themselves if this causes regressions. Idk.

Now how to do that, I'm not sure.

Reply:
I _think_ the code _appends_ new paths, which means adding duplicates doesn't change the search order. Which means not adding the duplicates doesn't change the search order either.

But it would be good to have it verified / confirmed.

Reply:
Yes, new paths are being append to the end. As long as the compiler searches the environment variable from start to end, it shouldn't change the detection order.